### ISSUE TITLE
Add first-class support for blurring what's underneath a Rectangle

### ISSUE THREAD

**Issue by tronical** [slint-ui/slint#2066]
Similar to `drop-shadow-*` properties we should have `blur-background-*` properties (or similarly named), to apply a blur effect to what's underneath.

This is not quite a sub-tree effect but should apply to whatever was rendered "before". This is expensive in the implementation (so needs to be used sparingly), but in that shape it would be super convenient for designers (this issue is based on that feedback).

Connected conceptually to #612.

**Comment by tronical** [comment]
Unfortunately no news yet.

**Comment by flukejones** [comment]
We would like to have this at JTE for our front panel.

**Comment by vgarleanu** [comment]
What would it take to implement this feature? What sort of difficulty?

**Comment by tronical** [comment]
If we say the this property applies to only what’s rendered exactly underneath the rectangle it’s used on, then I think this could be implemented by means of rendering the tree of items again starting at the root but (1) clipped to the rectangle and (2) not descending into the rectangle itself. This second pass would render into a layer and the blur would have to be applied there.

But I think this may the wrong approach.

Mayhe the existence of a blur in the tree means that we need to separate the scene into multiple layers (think multiple blurs) and the compose the layers accordingly. This avoids a double pass of the tree at the expense of more texture memory.

**Comment by tronical** [comment]
An alternative approach that would probably work with Skia out of the box is to perform a read-back of what's underneath into an offscreen surface. Skia's SaveLayerRec allows for exactly that, including the application of an image filter.

**Comment by Anivie** [comment]
In recent years, UI with blurred backgrounds has been gradually gaining popularity, and it seems to be becoming more common to see components using blur backgrounds.
Do we have any plans for this feature?

**Comment by FaintentDev** [comment]
> In recent years, UI with blurred backgrounds has been gradually gaining popularity, and it seems to be becoming more common to see components using blur backgrounds. Do we have any plans for this feature?

i second thee

**Comment by Siydge** [comment]
> In recent years, UI with blurred backgrounds has been gradually gaining popularity, and it seems to be becoming more common to see components using blur backgrounds. Do we have any plans for this feature?

Agreed, much of the software and UI I design utilizes subtle background blur because it adds a "polished" flair and makes software look more open in terms of cleanliness. Having the built in capability to blur backgrounds would be awesome, especially applied to windows themselves.

**Comment by Undead34** [comment]
any news about this?

**Comment by uiriansan** [comment]
Backdrop blur, cross application drag n' drop, [dynamic component loading](https://github.com/slint-ui/slint/discussions/959) and a proper built-in Video component are the only things holding me back from choosing Slint over Qt (cxx-qt) for my new project. It seems drag and drop is close to [being reality](https://github.com/rust-windowing/winit/pull/4571), and I guess I can use the [Image hack](https://github.com/slint-ui/slint/discussions/5463) for videos for now. I can think of a couple ways around the dynamic components, but I can't image my app without blur. Are these features on the roadmap? Can we expect them anytime soon?

**Comment by eira-fransham** [comment]
We've been talking about implementing arbitrary shaders for a while, with blur just being a special case, but it's a big topic. I made the suggestion the other day that blur is common enough that it would be worth special-casing (and I'm probably not the first to say so), others seemed to agree with me but whether that translates to it coming up soon on the roadmap is a different question.

**Comment by listepo** [comment]
### Use case: “Liquid Glass” / frosted material (not only window blur)

Building product UI that targets Apple-style **Liquid Glass** / Fluent Acrylic-like chrome (sidebars, sheets, floating toolbars) currently isn’t expressible in `.slint` without native hacks.

**What exists today**
- `opacity`, `drop-shadow-*`, solid/transparent fills
- Window-level blur only via private/platform hooks (`i-slint-backend-winit`, `set_system_backdrop` / `with_blur`) — not element-scoped, not portable in `.slint`

**What’s missing for a real glass material** (building on this issue + #612)

1. **Element backdrop blur** — blur pixels *under* a clipped `Rectangle` / layer (the core of this issue). Suggested API sketch:
   ```slint
   Rectangle {
     backdrop-blur: 16px;          // or blur-background: 16px;
     background: #ffffff22;
     border-width: 1px;
     border-color: #ffffff33;
   }
   ```

2. **Backdrop saturate / brightness** (optional, CSS-like) — glass rarely is blur alone: `backdrop-filter: blur(16px) saturate(160%)`

3. **Layer / offscreen subtree** (#612) — so glass panels, sheets, and popovers can composite without hand-rolling backends

4. **Public window material API** (separate but related) — Mica/Acrylic/vibrancy as a first-class `Window` property, not private winit hooks. Window blur ≠ in-scene glass.

5. **Accessibility hooks** — honor OS “reduce transparency” / high contrast by falling back to opaque fills when backdrop effects are active (designers need a documented path, even if automatic)

**Prior art**
- CSS `backdrop-filter`
- SwiftUI `Material` / Liquid Glass
- Fluent Acrylic (transient) + Mica (window base)
- Flutter `BackdropFilter` + `ImageFilter.blur`

**Why special-case blur first**  
Agree with comments here that full custom shaders are a larger topic; a curated `backdrop-blur` (+ optional saturate) unblocks the majority of “glass” UIs.

Happy to refine the API sketch or provide minimal repros/mock targets if useful.
